### PR TITLE
Update libidn2 to 2.2.0

### PR DESCRIPTION
**Pull request by Mno-hime**
**Testing**
- there's a fail in new test, reported as https://gitlab.com/libidn/libidn2/issues/72
- rebuild and run tests of `libidn2` dependents:
- [x] library/gnutls-3
- [x] mail/mutt
- [x] network/asterisk
- [x] web/curl
- [x] web/wget
- [x] [ABI Tracker](https://abi-laboratory.pro/index.php?view=timeline&l=libidn2): clean
- `wget` functional test

```
$ wget http://www.möhr.de/
--2019-05-24 07:46:08--  http://www.xn--mhr-sna.de/
Resolving www.xn--mhr-sna.de (www.xn--mhr-sna.de)... 62.144.206.150
Connecting to www.xn--mhr-sna.de (www.xn--mhr-sna.de)|62.144.206.150|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1093 (1.1K) [text/html]
Saving to: ‘index.html’
```
- `curl` functional test
```
$ curl -v http://www.möhr.de/
*   Trying 62.144.206.150...
* TCP_NODELAY set
* Failed to set TCP_KEEPALIVE on fd 4
* Connected to www.möhr.de (62.144.206.150) port 80 (#0)
> GET / HTTP/1.1
> Host: www.xn--mhr-sna.de
> User-Agent: curl/7.61.1
> Accept: */*
> 
< HTTP/1.1 200 OK
...
```

### CODE REVIEW

File: components/library/libidn2/libidn2.p5m [hunk 1]
```diff
@@ -26,14 +26,13 @@ license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
...
+link path=usr/lib/$(MACH64)/libidn2.so target=libidn2.so.0.3.6
+link path=usr/lib/$(MACH64)/libidn2.so.0 target=libidn2.so.0.3.6
+file path=usr/lib/$(MACH64)/libidn2.so.0.3.6
 file path=usr/lib/$(MACH64)/pkgconfig/libidn2.pc
-link path=usr/lib/libidn2.so target=libidn2.so.0.3.5
-link path=usr/lib/libidn2.so.0 target=libidn2.so.0.3.5
-file path=usr/lib/libidn2.so.0.3.5
+link path=usr/lib/libidn2.so target=libidn2.so.0.3.6
+link path=usr/lib/libidn2.so.0 target=libidn2.so.0.3.6
+file path=usr/lib/libidn2.so.0.3.6
```
Comment (pyhalov) [inline components/library/libidn2/libidn2.p5m:35]: What is its SONAME?

Comment (Mno-hime) [reply components/library/libidn2/libidn2.p5m:35]:
```
$ objdump -p /usr/lib/libidn2.so.0.3.6 | grep SONAME
  SONAME               libidn2.so.0
```
Why? Don't like it being zero?

Comment (pyhalov) [reply components/library/libidn2/libidn2.p5m:35]: I'm just asking because ABI tracker mentioned that they have occasionally increased SONAME last time.

Comment (Mno-hime) [reply components/library/libidn2/libidn2.p5m:35]: Yes, they botched it last time. We may wait for 2.2.0 to appear in the ABI Tracker to be sure about this and other similar problems.

Comment (Mno-hime) [reply components/library/libidn2/libidn2.p5m:35]: ABI Tracker is for libidn2 2.2.0 is published and is clean.

**Comment by Mno-hime** [comment]
> there's a fail in new test, reported as https://gitlab.com/libidn/libidn2/issues/72

This is failing everywhere but Linux. Not a problem for us.

**Pull request merged**